src/inner.jl

```diff
@@ -0,0 +1,143 @@
+using ITensors: inner, scalar, loginner
+using LinearAlgebra: norm, norm_sqr
+
+default_algorithm(tns::Tuple) = "bp"
+
+function ITensors.inner(
+  ϕ::AbstractITensorNetwork,
+  ψ::AbstractITensorNetwork;
+  alg=default_algorithm((ϕ, ψ)),
+  kwargs...,
+)
+  return inner(Algorithm(alg), ϕ, ψ; kwargs...)
+end
+
+function ITensors.inner(
+  ϕ::AbstractITensorNetwork,
+  A::AbstractITensorNetwork,
+  ψ::AbstractITensorNetwork;
+  alg=default_algorithm((ϕ, A, ψ)),
+  kwargs...,
+)
+  return inner(Algorithm(alg), ϕ, A, ψ; kwargs...)
+end
+
+function ITensors.inner(
+  alg::Algorithm"exact",
+  ϕ::AbstractITensorNetwork,
+  ψ::AbstractITensorNetwork;
+  sequence=nothing,
+  contraction_sequence_kwargs=(;),
+  kwargs...,
+)
+  tn = inner_network(ϕ, ψ; kwargs...)
+  if isnothing(sequence)
+    sequence = contraction_sequence(tn; contraction_sequence_kwargs...)
+  end
+  return scalar(tn; sequence)
+end
+
+function ITensors.inner(
+  alg::Algorithm"exact",
+  ϕ::AbstractITensorNetwork,
+  A::AbstractITensorNetwork,
+  ψ::AbstractITensorNetwork;
+  sequence=nothing,
+  contraction_sequence_kwargs=(;),
+  kwargs...,
+)
+  tn = inner_network(ϕ, A, ψ; kwargs...)
+  if isnothing(sequence)
+    sequence = contraction_sequence(tn; contraction_sequence_kwargs...)
+  end
+  return scalar(tn; sequence)
+end
+
+function ITensors.loginner(
+  ϕ::AbstractITensorNetwork,
+  ψ::AbstractITensorNetwork;
+  alg=default_algorithm(ϕ, ψ),
+  kwargs...,
+)
+  return loginner(Algorithm(alg), ϕ, ψ; kwargs...)
+end
+
+function ITensors.loginner(
+  ϕ::AbstractITensorNetwork,
+  A::AbstractITensorNetwork,
+  ψ::AbstractITensorNetwork;
+  alg=default_algorithm(ϕ, ψ),
```

Comment:
```suggestion
  alg=default_algorithm(ϕ, A, ψ),
```
Seems like it should allow accounting for `A` as well.